tests/components/motioneye/test_config_flow.py

```diff
@@ -488,13 +488,13 @@ async def test_advanced_options(hass: HomeAssistant) -> None:
             user_input={
                 CONF_WEBHOOK_SET: True,
                 CONF_WEBHOOK_SET_OVERWRITE: True,
-                CONF_STREAM_URL_TEMPLATE: "http://moo",
+                CONF_STREAM_URL_TEMPLATE: "",
             },
         )
         await hass.async_block_till_done()
         assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
         assert result["data"][CONF_WEBHOOK_SET]
         assert result["data"][CONF_WEBHOOK_SET_OVERWRITE]
-        assert result["data"][CONF_STREAM_URL_TEMPLATE] == "http://moo"
+        assert result["data"][CONF_STREAM_URL_TEMPLATE] == ""
```

Comment:
```suggestion
        assert result["data"][CONF_STREAM_URL_TEMPLATE] is None
```

We need to test for both. First check if it is `None` and then check if it is string with `isinstance`. You will need to call this function again to change the `CONF_STREAM_URL_TEMPLATE`.

```python
result = await hass.config_entries.options.async_configure(
    result["flow_id"],
    user_input={
        CONF_WEBHOOK_SET: True,         
        CONF_WEBHOOK_SET_OVERWRITE: True,
        CONF_STREAM_URL_TEMPLATE: "http://moo",
    },
)
await hass.async_block_till_done()
```

Reply:
Thank you again for your help. I've made the suggested changes to None but can I check...should the code look something like this? Or do I need to do the assertions twice?
```
        result = await hass.config_entries.options.async_configure(
            result["flow_id"],
            user_input={
                CONF_WEBHOOK_SET: True,
                CONF_WEBHOOK_SET_OVERWRITE: True,
                CONF_STREAM_URL_TEMPLATE: None,
            },
        )
        result = await hass.config_entries.options.async_configure(
            result["flow_id"],
            user_input={
                CONF_WEBHOOK_SET: True,
                CONF_WEBHOOK_SET_OVERWRITE: True,
                CONF_STREAM_URL_TEMPLATE: "http://moo",
            },
        )
        await hass.async_block_till_done()
        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
        assert result["data"][CONF_WEBHOOK_SET]
        assert result["data"][CONF_WEBHOOK_SET_OVERWRITE]
        assert result["data"][CONF_STREAM_URL_TEMPLATE] is None or result["data"][CONF_STREAM_URL_TEMPLATE] == "http://moo"
        assert len(mock_setup.mock_calls) == 0
        assert len(mock_setup_entry.mock_calls) == 0
```

Reply:
I realised that the above would be useless and two lots of assertations would be needed as the second `result` would overwrite the first `result`. I'm struggling with `isinstance` as I've not used that before - Would the below work instead? First checking the scenario where `CONF_STREAM_URL_TEMPLATE` is `None` and then checking when it is populated with `htto://moo` as was done previously?

```
        result = await hass.config_entries.options.async_configure(
            result["flow_id"],
            user_input={
                CONF_WEBHOOK_SET: True,
                CONF_WEBHOOK_SET_OVERWRITE: True,
                CONF_STREAM_URL_TEMPLATE: None,
            },
        )
        await hass.async_block_till_done()
        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
        assert result["data"][CONF_WEBHOOK_SET]
        assert result["data"][CONF_WEBHOOK_SET_OVERWRITE]
        assert result["data"][CONF_STREAM_URL_TEMPLATE] is None
        assert len(mock_setup.mock_calls) == 0
        assert len(mock_setup_entry.mock_calls) == 0

        result = await hass.config_entries.options.async_configure(
            result["flow_id"],
            user_input={
                CONF_WEBHOOK_SET: True,
                CONF_WEBHOOK_SET_OVERWRITE: True,
                CONF_STREAM_URL_TEMPLATE: "http://moo",
            },
        )
        await hass.async_block_till_done()
        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
        assert result["data"][CONF_WEBHOOK_SET]
        assert result["data"][CONF_WEBHOOK_SET_OVERWRITE]
        assert result["data"][CONF_STREAM_URL_TEMPLATE] == "http://moo"
        assert len(mock_setup.mock_calls) == 0
        assert len(mock_setup_entry.mock_calls) == 0
```

Reply:
Yes this will work instead 😃